[ios]update add-to-SwiftUI-app doc (#10888)
flutter/website: src/content/add-to-app/ios/add-flutter-screen.md

{% tabs "darwin-framework" %}
{% tab "SwiftUI" %}

> In this example, we create a `FlutterEngine` object inside a SwiftUI [Observable](https://developer.apple.com/documentation/observation/observable) object called `FlutterDependencies`. 

Reviewer: Hey as part of this change could you tweak [line 11](https://github.com/flutter/website/pull/10888/files#diff-b4b869d276ce91432a0be926d482945629735ae6a5fdb68eb6c9a1f5190d9054R11)?

```diff
- To launch a Flutter screen from an existing iOS, you start a
+ To launch a Flutter screen from an existing iOS app, you start a

Author: done

> In this example, we create a `FlutterEngine` object inside a SwiftUI [Observable](https://developer.apple.com/documentation/observation/observable) object called `FlutterDependencies`. 
> We pre-warm the engine by calling `run()`, and then inject this object into a `ContentView` using the `environment()` view modifier. 

Reviewer 2: ```suggestion
This example creates a `FlutterEngine` object inside a SwiftUI [`Observable`][]. 
Pre-warm the engine by calling `run()`, and then inject this object
into a `ContentView` using the `environment()` view modifier. 

[`Observable`]: https://developer.apple.com/documentation/observation/observable) object called `FlutterDependencies
```

{% tabs "darwin-framework" %}
{% tab "SwiftUI" %}

In this example, we create a `FlutterEngine` object inside a SwiftUI [`Observable`][] 
object called `FlutterDependencies`. 
Pre-warm the engine by calling `run()`, and then inject this object 
into a `ContentView` using the `environment()` view modifier. 

>  ```swift title="MyApp.swift"

Reviewer: About half the content is the normal app boilerplate, and the other half is Flutter additions. It is easy to miss additions.

For example, the engine is injected using the `environment(...)` view modifier. This is nested in the `Scene` declaration. Forgetting to add this won’t result in an error until subsequent steps.

Should we make the code samples in this page diffs? That would make it easier which lines need to be added for Flutter.

import SwiftUI
import Flutter

> // The following library connects plugins with iOS platform code to this app.

Reviewer: Super nitpick, consider:

```suggestion
// The following library contains plugins with iOS platform code to this app.
```

@main
struct MyApp: App {

>     // flutterDependencies will be injected via environment.

Reviewer: ```suggestion
    // flutterDependencies will be injected through the view environment.
```

Author: done

@main
struct MyApp: App {
    // flutterDependencies will be injected through the view environment.

>     @State var flutterDependencies = FlutterDependencies()

Reviewer: We seem to be inconsistent on this page whether we add a new line between a `struct` declaration and its first member. Could you do a pass on this page and double-check everything matches the recommended Swift style guide? (It's possible I just don't understand the Swift style, if so please ignore this feedback!)

{% endtab %}
{% tab "UIKit-ObjC" %}

> As an example, we demonstrate creating a FlutterEngine, exposed as a property, on app startup in the app delegate.

Author: This section is for UIKit-Swift, but we accidentally put SwiftUI content here. So I simply copied content from UIKit-ObjC.

> As an example, we demonstrate creating a FlutterEngine, exposed as a property, on app startup in the app delegate.

Reviewer 2: ```suggestion
The following example demonstrate creating a `FlutterEngine`
exposed as a property, on app startup in the app delegate.
```

{% tabs "darwin-framework" %}
{% tab "SwiftUI" %}

> The following example shows a generic `ContentView` with a `NavigationLink` hooked to a flutter screen. 
> We firstly create a `FlutterViewControllerRepresentable` to represent the `FlutterViewController`. 
> The `FlutterViewController` constructor takes the pre-warmed `FlutterEngine` as an argument, which is injected via environment. 

Reviewer 2: ```suggestion
The following example shows a generic `ContentView` with a
`NavigationLink` hooked to a flutter screen. 
First, create a `FlutterViewControllerRepresentable` to represent the
`FlutterViewController`. The `FlutterViewController` constructor
takes the pre-warmed `FlutterEngine` as an argument, which is injected through
the `environment` property. 
```

Reviewer: Also, consider linking `NavigationLink` to https://developer.apple.com/documentation/swiftui/navigationlink

Author: done and done

{% tabs "darwin-framework" %}
{% tab "SwiftUI" %}

The following example shows a generic `ContentView` with a 
[`NavigationLink`][] hooked to a flutter screen. 
First, create a `FlutterViewControllerRepresentable` to represent the 
`FlutterViewController`. The `FlutterViewController` constructor takes 
the pre-warmed `FlutterEngine` as an argument, which is injected through
the view environment. 

```swift title="ContentView.swift"
import SwiftUI
import Flutter

struct FlutterViewControllerRepresentable: UIViewControllerRepresentable {
  // Flutter dependencies are passed in through the view environment.
  @Environment(FlutterDependencies.self) var flutterDependencies
  
  func makeUIViewController(context: Context) -> some UIViewController {
    return FlutterViewController(
      engine: flutterDependencies.flutterEngine,
      nibName: nil,
      bundle: nil)
  }
  
  func updateUIViewController(_ uiViewController: UIViewControllerType, context: Context) {}
}

struct ContentView: View {

  var body: some View {
    NavigationStack {
      NavigationLink("My Flutter Feature") {
        FlutterViewControllerRepresentable()
      }

>     }

Reviewer: At the end of this section is this note:

![image](https://github.com/user-attachments/assets/e5049d47-c7e9-4cb5-8fea-3403f9b87aa0)

This note is incorrect for the `SwiftUI` tab (the `FlutterEngine` `run` method is called when the `FlutterDependencies` observable is initialized). Consider moving this note into the `UIKit-Swift` and `UIKit-ObjC` tabs.

Creating a subclass of the `FlutterAppDelegate` in UIKit apps was shown 
in the [Start a FlutterEngine and FlutterViewController section][]. 
In a SwiftUI app, you can create a subclass of the 

> `FlutterAppDelegate` and annotate it with the [`Observable()`](https://developer.apple.com/documentation/observation/observable()) macro as follows:

Reviewer 2: ```suggestion
`FlutterAppDelegate` and annotate it with the [`Observable`][] macro, as follows:
```

Author: done

Creating a subclass of the `FlutterAppDelegate` in UIKit apps was shown 
in the [Start a FlutterEngine and FlutterViewController section][]. 
In a SwiftUI app, you can create a subclass of the 
`FlutterAppDelegate` and annotate it with the [`Observable()`][] macro as follows:

> ```swift

Reviewer: ```suggestion
```swift title="MyApp.swift"
```

}
```

> Then, in your view, the `AppDelegate` is accessible via environment.

Reviewer 2: ```suggestion
In your view, access the `AppDelegate` through the `environment` property.
```

Author: updated "via environment" => "through the view environment", since environment not the name of the property

import Foundation
import Flutter

@Observable
class AppDelegate: UIResponder, UIApplicationDelegate, FlutterAppLifeCycleProvider {

  private let lifecycleDelegate = FlutterPluginAppLifeCycleDelegate()

>   let flutterEngine = FlutterEngine(name: "my flutter engine")

Author: `flutter_nps_engine` was probably from one of the earlier demo project. Update it to be consistent with other part of this doc.
